cmd/pmm-dump/util.go

```
@@ -100,18 +104,42 @@ func getPMMVersion(pmmURL string, c *client.Client) (string, string, error) {
 	return resp.Server.Version, resp.Server.FullVersion, nil
 }
 
-func getPMMServices(pmmURL string, c *client.Client) ([]dump.PMMServerService, error) {
+func getMajorVer(vers string) (*version.Version, error) {
+	v1, err := version.NewVersion(vers)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get major version")
+	}
+	return v1, nil
+}
+
+func isVer3(major *version.Version) bool {
+	constraints, err := version.NewConstraint("< 3.0.0")
+	if err != nil {
+		panic(fmt.Sprintf("cannot create constraint: %v", err))
+	}
+	return constraints.Check(major)
+}
+
+func getPMMServices(pmmURL string, c *client.Client, majorVersion *version.Version) ([]dump.PMMServerService, error) {
 	type servicesResp map[string][]struct {
 		ID     string `json:"service_id"`
 		Name   string `json:"service_name"`
 		NodeID string `json:"node_id"`
 	}
 
 	// Services
-
-	statusCode, body, err := c.Post(pmmURL + "/v1/inventory/Services/List")
+	var (
+		statusCode int
+		body       []byte
+		err        error
+	)
+	if isVer3(majorVersion) {
+		statusCode, body, err = c.Post(pmmURL + "/v1/inventory/Services/List")
+	} else {
+		statusCode, body, err = c.Get(pmmURL + "/v1/inventory/services")
+	}
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to get services: %d", statusCode)
```

Shall we return the error as well?

---

Fixed